### PR TITLE
feat(apply): apply kcl manifests

### PR DESCRIPTION
**Pull request by ekkoful**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [x] N
- [ ] Y 

#### 2. What is the scope of this PR (e.g. component or file name):

/pkg/options/apply.go

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [x] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: cmd/apply.go [hunk 1]
```diff
@@ -1,19 +1,27 @@
...
-
 	"github.com/spf13/cobra"
+	"kusionstack.io/kubectl-kcl/pkg/options"
 )
 
 // NewApplyCmd returns the run command.
 func NewApplyCmd() *cobra.Command {
-	return &cobra.Command{
-		Use:   "apply",
+	o := options.NewApplyOptions()
```
Comment (Peefy) [inline cmd/apply.go:10]: Good Job! Would U pls add more unit test cases or e2e test cases?

File: pkg/client/client.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+func GetGeneralResources(flags *genericclioptions.ConfigFlags, w io.Writer) error {
+	r, err := fetchResourcesBulk(flags)
+	if err != nil {
+		return err
+	}
+	p := printers.NewTypeSetter(scheme.Scheme).ToPrinter(&printers.YAMLPrinter{})
+	if err := p.PrintObj(r, w); err != nil {
+		panic(err.Error())
+	}
+	// fmt.Println(r)
```
Comment (Peefy) [inline pkg/client/client.go:23]: Remove un-used code.

File: examples/kcl-apply.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+functionConfig:
```
Comment (Peefy) [inline examples/kcl-apply.yaml:1]: The field `functionConfig:` is required? May I omit it?

File: pkg/client/client.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+// GetGeneralResources get kubernetes general resource like `kubectl get all`
+func GetGeneralResources(flags *genericclioptions.ConfigFlags, w io.Writer) error {
+	r, err := fetchResourcesBulk(flags)
+	if err != nil {
+		return err
+	}
+	p := printers.NewTypeSetter(scheme.Scheme).ToPrinter(&printers.YAMLPrinter{})
+	if err := p.PrintObj(r, w); err != nil {
+		panic(err.Error())
```
Comment (Peefy) [inline pkg/client/client.go:21]: Why panic here?

File: examples/kcl-apply.yaml (hunk 3 above)
Comment (ekkoful) [reply examples/kcl-apply.yaml:1]: Must need the field `functionConfig` from my test.

File: pkg/client/client_test.go
Comment (ekkoful) [inline pkg/client/client_test.go:1]: The unit test need the fake kubernetes server. So comment it. @Peefy

File: pkg/client/client.go (hunk 4 above)
Comment (ekkoful) [reply pkg/client/client.go:21]: Oh, Sorry. Debug run code test.

File: pkg/client/client_test.go
Comment (Peefy) [reply pkg/client/client_test.go:1]: > The unit test need the fake kubernetes server. So comment it. @Peefy

OK, we can use k3s as the fake k8s server.

Comment (ekkoful) [reply pkg/client/client_test.go:1]: OK. Let me try the k3s for test. May be the pr can merge first, if code have no problem after code review.

> > The unit test need the fake kubernetes server. So comment it. @Peefy
> 
> OK, we can use k3s as the fake k8s server.

Comment (ekkoful) [reply pkg/client/client_test.go:1]: image

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**